fedimint-core/src/api.rs

```diff
@@ -180,10 +180,28 @@ pub trait FederationApiExt: IFederationApi {
     /// Make an aggregate request to federation, using `strategy` to logically
     /// merge the responses.
     async fn request_with_strategy<MemberRet: serde::de::DeserializeOwned, FedRet: Debug>(
+        &self,
+        strategy: impl QueryStrategy<MemberRet, FedRet> + MaybeSend,
+        method: String,
+        params: ApiRequestErased,
+    ) -> FederationResult<FedRet> {
+        self.request_with_strategy_with_timeout(strategy, method, params, None)
+            .await
+    }
+
+    /// Make an aggregate request to federation, using `strategy` to logically
+    /// merge the responses.
+    ///
+    /// Optionally put a timeout on every request.
```

Comment:
The implicit result might be that the request might run forever, right? Because different peers may time out and then work again, never leading to complete failure but also not to success.

Reply:
Yes.

Now that mention it, I think maybe the timeout should be provided by a strategy, and not a separate argument. Because it only makes sense in the context of a specific strategy that relies on it...